Sync USERNAME_REGEXP with the UI one

LazarenkoDmytro opened the pull request:
An inconsistency existed between the UI's regular expression and the backend’s one. This inconsistency has been rectified.

The UI's regex:
`NamePattern: /^[ґҐіІєЄїЇА-Яа-яa-zA-Z](?!.*\.$)(?!.*?\.\.)(?!.*?--)(?!.*?'')[-'ʼ’ ґҐіІєЄїЇА-Яа-я\w.]{0,29}$/`

LazarenkoDmytro committed:
Sync USERNAME_REGEXP with the UI one

LazarenkoDmytro committed:
Remove invalid test arguments from DTO test cases.

Previously, these test cases were necessary. However, since the requirement to capitalize the first letter has been removed, the corresponding test cases have been eliminated.

dxrknesss reviewed (approved):
LGTM

RomaMocherniuk commented on service-api/src/main/java/greencity/constant/ValidationConstants.java:
```diff
@@ -25,11 +25,5 @@ public class ValidationConstants {
-            {1,30}\
-            (?<![ЭэЁёъЪЫы])$\
-            """;
+        "^[ґҐіІєЄїЇА-Яа-яa-zA-Z](?!.*\\.$)(?!.*?\\.\\.)(?!.*?--)(?!.*?'')[-'ʼ’ ґҐіІєЄїЇА-Яа-я\\w.]{0,29}$";
```
Please add tests for all cases
it seems like some cases are not covered

LazarenkoDmytro committed:
Expand test cases for DTO validation.

Added more comprehensive valid and invalid test cases for various DTOs to improve coverage and validation robustness. This includes additional name formats, edge cases, and boundary scenarios for better reliability.
